AK/BufferedStream.h

```diff
@@ -87,7 +87,8 @@ class BufferedHelper {
         auto const candidate = TRY(find_and_populate_until_any_of(candidates, buffer.size()));
 
         if (stream().is_eof()) {
-            if (buffer.size() < m_buffer.used_space()) {
+            if ((candidate.has_value() && candidate->offset + candidate->size > buffer.size())
```

Comment:
What's the relation between this and `stream().is_eof()`? Why is it only error-worthy that the delimiter is behind the readable buffer size _after_ the source stream is EOF?

Reply:
I have no idea honestly. I'm just assuming that the check is reasonable based on the comment below, and extending the check to the full set of possible return values. @sin-ack had some comments in discord but I didn't get the feeling that we came to a conclusion.

Reply:
Discussed on discord, we should probably extend this check to the non-eof case as well, but that will need some more testing. Added a test as well.